FV3/atmos_cubed_sphere/tools/fv_mp_mod.F90

```diff
@@ -635,14 +643,49 @@ subroutine domain_decomp(npx,npy,nregions,grid_type,nested,Atm,layout,io_layout)
                   tile_id(n) = n
                enddo
             endif
+            allocate(xextent(layout(1),nregions), yextent(layout(2),nregions))
+            xextent(:, :) = 0
+            yextent(:, :) = 0
+            if (shrink_factor < 1.0) then
+                ! We reduce the width of the subdomains at the edges so that they are edge_subdomain_shrink_factor
+                ! smaller than "inner" subdomains with no edges/corners. We compute the size of the edge subdomains
+                ! in shrink_size and then evenly distribute the remaining gridpoints using mpp_compute_extent().
+               if (layout(1) > 2) then
+                  allocate(ibegin(layout(1)-2), iend(layout(1)-2))
+                  shrink_size = nint( real(nx) / (2.0 + real(layout(1)-2)/shrink_factor) )
```

Comment:
For some reason I'm not following this particular line. I'm sure it's correct, but I'd appreciate sitting down to go over it if possible

Reply:
I've added a comment that explains the logic in the comment of the new subroutine that does the computation. Can you take a look and let me know if it's clear now?